### PR TITLE
Adding enos_command module and unit test

### PR DESCRIPTION
**Pull request by amuraleedhar**
SUMMARY

Creating the command module for enos from Lenovo.

ISSUE TYPE

New Module Pull Request

COMPONENT NAME

lib/ansible/modules/network/enos/enos_command.py
test/units/modules/network/enos/test_enos_command.py

ANSIBLE VERSION

ansible 2.5.0
config file = /etc/ansible/ansible.cfg
configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible-2.5.0-py2.7.egg/ansible
executable location = /usr/local/bin/ansible
python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

ADDITIONAL INFORMATION

This is part of an effort to create modules for enos based switches from Lenovo.

### CODE REVIEW

**Comment by amuraleedhar** [comment]
Please review these and merge it .
Thanks 
-Anil

**Comment by amuraleedhar** [comment]
Please review the enos_command module and merge it .
Thanks
-Anil

File: lib/ansible/modules/network/enos/enos_command.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+---
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    port: 22
+    username: admin
+    password: admin
+    timeout: 30
+    authorize: True
+    auth_pass:
```
Comment (ganeshrn) [inline lib/ansible/modules/network/enos/enos_command.py]: Value is missing here.

File: lib/ansible/modules/network/enos/enos_command.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        retries=dict(default=10, type='int'),
+        interval=dict(default=1, type='int')
+    )
+
+    spec.update(enos_argument_spec)
+
+    module = AnsibleModule(argument_spec=spec, supports_check_mode=True)
+
+    warnings = list()
+    check_args(module, warnings)
```
Comment (ganeshrn) [inline lib/ansible/modules/network/enos/enos_command.py]: Is `check_args` call required, as the function is empty? Fo other network modules it is present due to legacy reasons which need to removed in respective platform modules refactor.

File: lib/ansible/modules/network/enos/enos_command.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_command.py]: Removing auth_pass as its not required by default.

File: lib/ansible/modules/network/enos/enos_command.py (hunk 2 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_command.py]: Will remove that

**Review by ganeshrn — APPROVED** [review]
lgtm

File: lib/ansible/modules/network/enos/enos_command.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  wait_for:
+    description:
+      - List of conditions to evaluate against the output of the
+        command. The task will wait for each condition to be true
+        before moving forward. If the conditional is not true
+        within the configured number of retries, the task fails.
+        See examples.
+    required: false
+    default: null
+    aliases: ['waitfor']
```
Comment (gundalow) [inline lib/ansible/modules/network/enos/enos_command.py]: Please remove this line. This is only in old modules for backwards compatibility.

File: lib/ansible/modules/network/enos/enos_command.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            item = str(item).split('\n')
+        yield item
+
+
+def main():
+    spec = dict(
+        # { command: <str>, prompt: <str>, response: <str> }
+        commands=dict(type='list', required=True),
+
+        wait_for=dict(type='list', aliases=['waitfor']),
```
Comment (gundalow) [inline lib/ansible/modules/network/enos/enos_command.py]: Please remove `, aliases=['waitfor']`

File: lib/ansible/modules/network/enos/enos_command.py [hunk 5]
```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python
+#
+# Copyright: Ansible Project
```
Comment (gundalow) [inline lib/ansible/modules/network/enos/enos_command.py]: Lenovo copyright?

File: test/units/modules/network/enos/test_enos_command.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
+# (c) 2016 Red Hat Inc.
```
Comment (gundalow) [inline test/units/modules/network/enos/test_enos_command.py]: 2017
Lenovo copyright?

**Review by gundalow — CHANGES_REQUESTED** [review]
Just a few minor comments

File: test/units/modules/network/enos/test_enos_command.py (hunk 6 above)
Comment (amuraleedhar) [reply test/units/modules/network/enos/test_enos_command.py]: Yes, Changing it

File: lib/ansible/modules/network/enos/enos_command.py (hunk 5 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_command.py]: Yes Changing it.

File: lib/ansible/modules/network/enos/enos_command.py (hunk 3 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_command.py]: Removed

File: lib/ansible/modules/network/enos/enos_command.py (hunk 4 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/enos/enos_command.py]: Removed

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into `devel` will be released in `2.5.0`

**Pull request merged**